File: app/src/main/java/com/commit451/gitlab/viewHolder/AccountViewHolder.java
package com.commit451.gitlab.viewHolder;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.PopupMenu;
import android.widget.TextView;

import com.commit451.gitlab.App;
import com.commit451.gitlab.R;
import com.commit451.gitlab.model.Account;
import com.commit451.gitlab.transformation.CircleTransformation;
import com.commit451.gitlab.util.ImageUtil;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * A signed in account
 */
public class AccountViewHolder extends RecyclerView.ViewHolder{

    public static AccountViewHolder inflate(ViewGroup parent) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_account, parent, false);
        return new AccountViewHolder(view);
    }

    @BindView(R.id.account_image) ImageView image;
    @BindView(R.id.account_username) TextView textUsername;
    @BindView(R.id.account_server) TextView textServer;
    @BindView(R.id.account_more) View buttonMore;

    public final PopupMenu mPopupMenu;

    public AccountViewHolder(View view) {
        super(view);
        ButterKnife.bind(this, view);

        mPopupMenu = new PopupMenu(itemView.getContext(), buttonMore);
        mPopupMenu.getMenuInflater().inflate(R.menu.item_menu_account, mPopupMenu.getMenu());

        buttonMore.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mPopupMenu.show();
            }
        });
    }

    public void bind(Account account, boolean isActive, int colorSelected) {
        textServer.setText(account.getServerUrl().toString());
        textUsername.setText(account.getUser().getUsername());

        if (isActive) {
            itemView.setBackgroundColor(colorSelected);
        } else {
            itemView.setBackground(null);
        }

        App.get().getPicasso()
                .load(ImageUtil.getAvatarUrl(account.getUser(), itemView.getResources().getDimensionPixelSize(R.dimen.user_list_image_size)))
                .transform(new CircleTransformation())
                .into(image);
    }
}

File: app/src/main/java/com/commit451/gitlab/viewHolder/AccountViewHolder.kt
package com.commit451.gitlab.viewHolder

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.PopupMenu
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import coil.api.load
import coil.transform.CircleCropTransformation
import com.commit451.addendum.recyclerview.bindView
import com.commit451.gitlab.R
import com.commit451.gitlab.model.Account
import com.commit451.gitlab.util.ImageUtil

/**
 * A signed in account
 */
class AccountViewHolder(view: View) : RecyclerView.ViewHolder(view) {

    companion object {

        fun inflate(parent: ViewGroup): AccountViewHolder {
            val view = LayoutInflater.from(parent.context)
                    .inflate(R.layout.item_account, parent, false)
            return AccountViewHolder(view)
        }
    }

    private val image: ImageView by bindView(R.id.account_image)
    private val textUsername: TextView by bindView(R.id.account_username)
    private val textServer: TextView by bindView(R.id.account_server)
    private val buttonMore: View by bindView(R.id.account_more)

    val popupMenu: PopupMenu

    init {
        popupMenu = PopupMenu(itemView.context, buttonMore)
        popupMenu.menuInflater.inflate(R.menu.logout, popupMenu.menu)

        buttonMore.setOnClickListener { popupMenu.show() }
    }

    fun bind(account: Account, isActive: Boolean, colorSelected: Int) {
        textServer.text = account.serverUrl.toString()
        textUsername.text = account.email

        if (isActive) {
            itemView.setBackgroundColor(colorSelected)
        } else {
            itemView.background = null
        }

        image.load(ImageUtil.getAvatarUrl(account.email, itemView.resources.getDimensionPixelSize(R.dimen.user_list_image_size))) {
            transformations(CircleCropTransformation())
        }
    }
}
